ape_starknet/ecosystems.py

```diff
@@ -98,31 +97,14 @@ def decode_returndata(self, abi: MethodABI, raw_data: List[int]) -> Any:  # type
         if not raw_data:
             return raw_data
 
-        raw_data = [self.encode_primitive_value(v) for v in raw_data]
-        iter_data = iter(raw_data)
-        decoded: List[Any] = []
+        full_abi = [
+            a.dict() for a in (abi.contract_type.abi if abi.contract_type is not None else [abi])
+        ]
```

Comment:
`contract_abi` should always be available at that point:

```suggestion
        full_abi = [a.dict() for a in abi.contract_type.abi]  # type: ignore
```

Reply:
mypy does not know that

Reply:
That's the purpose of`type: ignore`.